src/RandomUser.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from "react";
+import Button from "./Button";
+import UserProfile from "./UserProfile";
+
+const RandomUser = () => {
+  const [user, setUser] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const fectchUser = () => {
+    fetch(`https:randomuser.me/api/?results=5`);
+    setLoading(true)
+      .then((res) => {
```

Comment:
setLoading is not a promise with the fetch result, you might have moved the `setLoading(true)` one line down by mistake?